### PR TITLE
Separation of policies and user journeys

### PR DESCRIPTION
**Pull request by slorek**
**Context:**

Prior to the introduction of the LUPP policy and the combined ECP/LUPP user journey, it was assumed that each policy would have one distinct user journey.

This assumption proved untrue with the introduction of the combined ECP/LUPP journey, which was an iteration of the existing ECP journey so that eligibility for either policy could be established as both policies had similar questions and eligibility criteria. Therefore at the time, for expediancy, the LUPP policy was added to the existing ECP journey with minimal code changes.

Now, as well as realising that one user journey may cover more than one policy, we understand from current alpha investigations that some policies may require multiple user journeys.

**Problems**

* It's not clear how to add a new user journey as the code supporting this is not located together or clearly distinct from policies
* It is likely we will need the ability to add more than one user journey for one policy
* Policy models should only define eligibility criteria and policy-specific functionality, and not front end concerns

**Proposed Solution:**

Separate the concepts of user journeys and policies so that there is flexibility to accommodate the new scenarios identified. Document the steps needed to add a new journey to tease out other complexities.

**Changes completed:**

* The existing `PolicyConfiguration` model has been renamed `JourneyConfiguration` to more accurately reflect what it is used for.
* The existing LUPP/ECP combined journey views and translations have been moved from the `early_career_payments` namespace to a new `additional_payments` namespace so it is clear which journey they apply to, and to avoid confusion since this journey covers both ECP and LUPP policies.

**TO DO:**
* Detach slug sequences from policies
* Remove all front end concerns from Policy models etc.
* Fully segregate concerns for each user journey
* LOTS MORE TO DO...

### CODE REVIEW

**Review by your — COMMENTED** [review]
LGTM so far 👍

File: config/locales/en.yml [hunk 1]
```diff
@@ -342,16 +342,10 @@ en:
...
-    policy_acronym: "ECP"
-    payment_name: "Early-career payment"
+    journey_name: "Claim additional payments for teaching"
     claim_description: "for an additional payment for teaching"
-    claim_subject: "Early-career payment"
-    claim_amount_description: "Additional payment for teaching"
-    purpose: "For early career teachers who teach certain subjects."
-    support_email_address: "earlycareerteacherpayments@digital.education.gov.uk"
     landing_page: "Find out if you are eligible for any additional payments"
+    support_email_address: "additionalteachingpayment@digital.education.gov.uk"
```
[the remark this answers is not in the document]
Comment (slorek) [reply config/locales/en.yml:348]: Yes, copied from https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/master/config/locales/en.yml#L46.

**Review by Reyko — COMMENTED** [review]
Looks good, left a suggestion for the migration

File: db/migrate/20220523175052_change_policy_configuration_policy_type.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@ class ChangePolicyConfigurationPolicyType < ActiveRecord::Migration[6.0]
   def up
     add_column :policy_configurations, :policy_types, :text, array: true, default: []
 
-    PolicyConfiguration.reset_column_information
-    PolicyConfiguration.all.each do |pc|
+    JourneyConfiguration.reset_column_information
```
[the remark this answers is not in the document]
Comment (slorek) [reply db/migrate/20220523175052_change_policy_configuration_policy_type.rb:5]: Nice, thanks.

**Review by Reyko — APPROVED** [review]

**Review by your — APPROVED** [review]

**Pull request merged**